zcash_primitives/src/transaction/components/sapling/builder.rs

```diff
@@ -113,20 +114,19 @@ impl<P: consensus::Parameters> SaplingOutput<P> {
             to,
             note,
             memo: memo.unwrap_or_else(MemoBytes::empty),
-            _params: PhantomData::default(),
         })
     }
 
-    pub fn build<Pr: TxProver, R: RngCore + CryptoRng>(
+    pub fn build<P: consensus::Parameters, Pr: TxProver, R: RngCore + CryptoRng>(
```

Comment:
This would enable using different parameters to construct the builder and then build with it. Undo these changes and re-introduce `P: consensus::Parameters` as a bound on the builder.

Reply:
Actually, we could instead make `SaplingOutput` an internal type, and then these changes would be fine. `SaplingOutput` was public because we wanted to enable external transaction builders to add Sapling outputs, but now we have `SaplingBuilder` which provides that functionality with an improved API.